### PR TITLE
Developed Remove student, get multiple feedback, post feedback and get feedback routes

### PR DESCRIPTION
**Pull request by yashodgayashan**
## FYI

* Remove student route will remove the student only if he is not completed information.
* get multiple feedback route will either return all the feedback for a particular student or if given the batch it will return the feedback regarding that batch.
* post feedback will post a feedback.
* get feedback will get a single feedback.

### CODE REVIEW

File: BackendNew/controllers/students.js [hunk 1]
```diff
@@ -263,3 +262,38 @@ const updateStudentSelectCompanyInfo = (req, isUpdated) => {
...
+      } else {
+        // If no one is removed
+        if (result.data.affectedRows == 0) {
+          res
+            .status(400)
+            .send({ message: "There is no student with the batch" });
+        } else {
+          students.removeStudent(studentId, (err, result) => {
+            if (err) {
+              res.status(400).send({ message: "Cannot removed the student" });
```
Comment (hasinichathu) [inline BackendNew/controllers/students.js:284]: Will this rollback previous changes done to  table " StudentHasBatch"?

File: BackendNew/controllers/students.js [hunk 2]
```diff
@@ -263,3 +262,38 @@ const updateStudentSelectCompanyInfo = (req, isUpdated) => {
...
+
+    students.removeStudentHasBatch(studentId, batch, (err, result) => {
+      if (err) {
+        res.status(500).send();
+      } else {
+        // If no one is removed
+        if (result.data.affectedRows == 0) {
+          res
+            .status(400)
+            .send({ message: "There is no student with the batch" });
```
Comment (hasinichathu) [inline BackendNew/controllers/students.js:280]:
```suggestion
            .send({ message: "There is no student associated with this batch" });
```

File: BackendNew/models/feedback.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+// Import database connection
+const sql = require("./database.js");
+
+exports.getFeedbacks = (studentId, sendFeedbacks) => {
```
Comment (hasinichathu) [inline BackendNew/models/feedback.js]: It should be feedback

File: BackendNew/models/feedback.js [hunk 4]
```diff
@@ -12,3 +12,15 @@ exports.getFeedbacks = (studentId, sendFeedbacks) => {
     }
   });
 };
+
+exports.getFeedbacksByBatch = (studentId, batchId, sendFeedbacks) => {
```
Comment (hasinichathu) [inline BackendNew/models/feedback.js]:
```suggestion
exports.getFeedbackByBatch = (studentId, batchId, sendFeedback) => {
```

File: BackendNew/models/feedback.js [hunk 5]
```diff
@@ -12,3 +12,15 @@ exports.getFeedbacks = (studentId, sendFeedbacks) => {
...
+  var sqlString =
+    "SELECT FeedbackId, BatchId, IndexNumber FROM feedback WHERE IndexNumber = ? AND BatchId = ?";
+  sql.query(sqlString, [studentId, batchId], (err, result) => {
+    if (err) {
+      sendFeedbacks(err, null);
```
Comment (hasinichathu) [inline BackendNew/models/feedback.js:21]:
```suggestion
      sendFeedback(err, null);
```

File: BackendNew/models/feedback.js [hunk 6]
```diff
@@ -12,3 +12,15 @@ exports.getFeedbacks = (studentId, sendFeedbacks) => {
...
+    } else {
+      sendFeedbacks(null, { data: result });
```
Comment (hasinichathu) [inline BackendNew/models/feedback.js:23]:
```suggestion
      sendFeedback(null, { data: result });
```

File: BackendNew/controllers/feedback.js [hunk 7]
```diff
@@ -0,0 +1,23 @@
+const feedback = require("../models/feedback");
+
+exports.getmultipleFeedback = (req, res) => {
```
Comment (hasinichathu) [inline BackendNew/controllers/feedback.js:3]:
```suggestion
exports.getMultipleFeedback = (req, res) => {
```

File: BackendNew/controllers/students.js (hunk 1 above)
Comment (yashodgayashan) [reply BackendNew/controllers/students.js:284]: No. should we do that ?

Comment (hasinichathu) [reply BackendNew/controllers/students.js:284]: If error occurred at this stage there will be student who does not have a batch. Wont't this affect badly at code later?

Comment (yashodgayashan) [reply BackendNew/controllers/students.js:284]: What do you suggest for this?

**Pull request merged**